<a href="https://colab.research.google.com/github/carolinampessoa/TechChallengeFase1/blob/main/TechChallengeFase1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Importando bibliotecas



In [2]:
# Bibliotecas para leitura e carregamento de dataset do Kaggle
import kagglehub
import pandas as pd
import os
# Bibliotecas para visualização de dados
import seaborn as sns
import matplotlib.pyplot as plt
# Bibliotecas para ajuste de dados
from sklearn.preprocessing import StandardScaler
from sklearn.preprocessing import OneHotEncoder, StandardScaler
from sklearn.compose import ColumnTransformer
# Bibliotecas para treinamento de modelos
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.ensemble import RandomForestRegressor
from sklearn.tree import DecisionTreeRegressor
from sklearn.neighbors import KNeighborsRegressor
from xgboost import XGBRegressor
from sklearn.model_selection import GridSearchCV
# Bibliotecas para validação de modelos
from sklearn.metrics import mean_squared_error, r2_score, mean_absolute_error
import numpy as np
import statsmodels.api as sm

**Passo 1: Exploração de dados**

Baixando a base de dados e explorando suas características


In [6]:
path = kagglehub.dataset_download("mirichoi0218/insurance");
df = pd.read_csv(os.path.join(path, "insurance.csv"));

# Exibindo as primeiras linhas e formato do DataFrame
print(df.head());
print("Formato do dataset (linhas, colunas):", df.shape);

   age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

Formato do dataset (linhas, colunas): (1338, 7)


Base de dados de composta por 1338 registros e 7 colunas, sendo 6 variáveis de entrada (idade, genero, IMC, número de filhos, fumante e região) e uma váriavel de saída (custos - variável target).

In [7]:
# Características do DataFrame
print("\nInformações do DataFrame:");
print(df.info());  # Tipos de dados: categóricas(sex, smoker, region), inteiro (age, children) e float (BMI, charges).


Informações do DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB
None


 Tipos de dados presentes no dataset: categóricas (sex, smoker, region), inteiros (age, children) e float (BMI, charges).

 Não existem valores nulos na base de dados.

Deseja fazer a análise exploratória dos dados? (s/n)
n
Fitting 3 folds for each of 72 candidates, totalling 216 fits

Resultados dos Modelos (conjunto de treino):
              Modelo    R²     RMSE      MAE  MAPE (%)
0  Linear Regression  0.51  8391.12  4270.77     27.02
1      Decision Tree  1.00   554.21    29.57      0.50
2      Random Forest  0.96  2396.26   987.11      7.62
3                KNN  0.80  5357.56  2813.33     21.14
4            XGBoost  0.88  4181.86  1917.21     14.49

Resultados dos Modelos (conjunto de treino):
              Modelo    R²     RMSE      MAE  MAPE (%)
0  Linear Regression  0.58  8050.05  4011.62     28.86
1      Decision Tree  0.75  6171.27  2978.59     36.48
2      Random Forest  0.87  4413.00  2249.35     22.44
3                KNN  0.74  6326.59  3453.60     28.24
4            XGBoost  0.88  4310.42  1958.12     15.56


*texto em itálico*# Nova seção